### ISSUE TITLE
Upgrading React native from 0.74 to 0.76 in existing project creates Jittering effect

### ISSUE THREAD

**Issue by shreygargofficial** [facebook/react-native#47926]
### Description

I was using expo sdk 51.0.28 and RN 0.74.5 after upgrading it to latest expo and RN, I am facing lot of jittering effect in app 

**Earlier package.json**

`{
  "name": "vyayam",
  "version": "1.0.0",
  "main": "expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web"
  },
  "dependencies": {
    "@expo/vector-icons": "^14.0.4",
    "@hookform/resolvers": "^3.9.0",
    "@react-native-async-storage/async-storage": "^1.23.1",
    "@react-native-community/datetimepicker": "^8.2.0",
    "@react-native-community/slider": "4.5.2",
    "@react-native-picker/picker": "^2.8.1",
    "@react-navigation/bottom-tabs": "^6.6.1",
    "@react-navigation/drawer": "^6.7.2",
    "@react-navigation/native": "^6.1.18",
    "@react-navigation/native-stack": "^6.11.0",
    "@reduxjs/toolkit": "^2.2.8",
    "axios": "^1.7.7",
    "expo": "~51.0.28",
    "expo-font": "~12.0.10",
    "expo-linear-gradient": "^13.0.2",
    "expo-linking": "~6.3.1",
    "expo-secure-store": "^13.0.2",
    "expo-splash-screen": "~0.27.6",
    "expo-status-bar": "~1.12.1",
    "lottie-react-native": "6.7.0",
    "react": "18.2.0",
    "react-hook-form": "^7.53.0",
    "react-native": "0.74.5",
    "react-native-chart-kit": "^6.12.0",
    "react-native-gesture-handler": "~2.16.1",
    "react-native-htmlview": "^0.17.0",
    "react-native-paper": "^5.12.5",
    "react-native-reanimated": "~3.10.1",
    "react-native-safe-area-context": "4.10.5",
    "react-native-screens": "3.31.1",
    "react-native-svg": "^15.8.0",
    "react-native-webview": "13.8.6",
    "react-redux": "^9.1.2",
    "redux-persist": "^6.0.0",
    "redux-thunk": "^3.1.0",
    "yup": "^1.4.0"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0"
  },
  "private": true
}
`

**After Upgrade to latest one**

`{
  "name": "vyayam",
  "version": "1.0.0",
  "main": "expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web"
  },
  "dependencies": {
    "@expo/vector-icons": "^14.0.2",
    "@hookform/resolvers": "^3.9.0",
    "@react-native-async-storage/async-storage": "1.23.1",
    "@react-native-community/datetimepicker": "8.2.0",
    "@react-native-community/slider": "4.5.5",
    "@react-native-picker/picker": "2.9.0",
    "@react-navigation/bottom-tabs": "^6.6.1",
    "@react-navigation/drawer": "^6.7.2",
    "@react-navigation/native": "^6.1.18",
    "@react-navigation/native-stack": "^6.11.0",
    "@reduxjs/toolkit": "^2.2.8",
    "axios": "^1.7.7",
    "expo": "~52.0.11",
    "expo-font": "~13.0.1",
    "expo-linear-gradient": "~14.0.1",
    "expo-linking": "~7.0.3",
    "expo-secure-store": "~14.0.0",
    "expo-splash-screen": "~0.29.13",
    "expo-status-bar": "~2.0.0",
    "lottie-react-native": "7.1.0",
    "react": "18.3.1",
    "react-hook-form": "^7.53.0",
    "react-native": "0.76.3",
    "react-native-chart-kit": "^6.12.0",
    "react-native-gesture-handler": "~2.20.2",
    "react-native-htmlview": "^0.17.0",
    "react-native-paper": "^5.12.5",
    "react-native-reanimated": "~3.16.1",
    "react-native-safe-area-context": "4.12.0",
    "react-native-screens": "~4.1.0",
    "react-native-svg": "15.8.0",
    "react-native-webview": "13.12.2",
    "react-redux": "^9.1.2",
    "redux-persist": "^6.0.0",
    "redux-thunk": "^3.1.0",
    "yup": "^1.4.0"
  },
  "devDependencies": {
    "@babel/core": "^7.25.2"
  },
  "private": true
}
`

https://github.com/user-attachments/assets/ba5917e8-bf79-4e8e-a3bc-e704a4d9b1f9

https://github.com/user-attachments/assets/5fdd5771-853a-47b0-956a-87cdee25906f

### Steps to reproduce

https://github.com/shreygargofficial/vyayam/tree/upgrade

clone it do npm i then npm start

### React Native Version

0.76.3

### Output of `npx react-native info`

```text
⚠️ react-native depends on @react-native-community/cli for cli commands. To fix update your package.json to include:


  "devDependencies": {
    "@react-native-community/cli": "latest",
  }
```

### Screenshots and Videos

![Screenshot_1732532098](https://github.com/user-attachments/assets/8efd6580-5ce7-4b5f-9d91-1373ee5b6f0f)

**Comment by Fab-Diaz** [comment]
Agreed, I'm facing the same issue

**Comment by migueldaipre** [comment]
Hey @shreygargofficial, thanks for opening this issue.

This doesn't seem to be a issue with the react-native core. We need a stand-alone reproducer (react-native core only)

**Comment by migueldaipre** [comment]
@Fab-Diaz can you share your dependencies?

**Comment by LeXuanKhanh** [comment]
I have the same issue when creating a new project with the latest Expo 52. Whenever I click / hold the button or navigate to a new screen, it will cause flickering. Would it still count as an issue in react-native core ? 

https://github.com/user-attachments/assets/904cf4ee-cdc0-4922-861c-a47380014534

**Comment by Fab-Diaz** [comment]
Here another example between a modal and a new screen. 

https://github.com/user-attachments/assets/69a70cd9-0027-4e81-af9c-83c061fcd30c

Will try to isolate the issue, see which dependency / version is responsible

**Comment by Kudo** [comment]
the `https://github.com/shreygargofficial/vyayam/tree/upgrade` returns 404. could you help to check the repro is valid?

**Comment by shreygargofficial** [comment]
> the https://github.com/shreygargofficial/vyayam/tree/upgrade returns 404. could you help to check the repro is valid?

sorry, I made it private...If you want to check I can make it public

**Comment by Kudo** [comment]
> sorry, I made it private...If you want to check I can make it public

@shreygargofficial yes please, we would like to investigate the issue. would be good if you could make the repro public.

**Comment by Fab-Diaz** [comment]
Another more extreme example between a modal and a page

https://github.com/user-attachments/assets/eef0dfba-5206-4044-9b82-a22237c8df44

**Comment by shreygargofficial** [comment]
> > Another more extreme example between a modal and a page
> > 
> >   
> >     
> >     
> > 
> >     Screen.Recording.2024-12-04.at.11.26.48.mov
> 
> are you using expo as well ?

Yes

**Comment by shreygargofficial** [comment]
> > sorry, I made it private...If you want to check I can make it public
> 
> @shreygargofficial yes please, we would like to investigate the issue. would be good if you could make the repro public.

Done..Now it is public

**Comment by Kudo** [comment]
> Done..Now it is public

thanks @shreygargofficial. one thing i've noticed before digging is that you use old architecture on sdk 51 and `newArchEnabled=true` on sdk 52. that could have much difference between the old/new architecture. i guess i would  try to turn on new arch on sdk 51 or disable new arch on sdk 52 to see if the performance is still valid.

**Comment by shreygargofficial** [comment]
> > Done..Now it is public
> 
> thanks @shreygargofficial. one thing i've noticed before digging is that you use old architecture on sdk 51 and `newArchEnabled=true` on sdk 52. that could have much difference between the old/new architecture. i guess i would try to turn on new arch on sdk 51 or disable new arch on sdk 52 to see if the performance is still valid.

I tried that but it was not working

**Comment by Kudo** [comment]
@shreygargofficial  i can repro your issue.
- it works fine on sdk 51, both new arch and old arch.
- from sdk 52, the issue happens no matter on new arch or old arch.
- mostly relate to keyboard. from your two videos, i cannot clearly see whether you touched at any textinput? could you help to clarify this? the jitter issue happens when i touch textinput (and keyboard appears)
- the most trivial repro step for me is from login screen -> signup -> touch textinput -> back to login screen. as the video shown: [Screen_recording_20241205_024251.webm](https://github.com/user-attachments/assets/6cb85857-ad43-46ef-835a-027e60abd400)

will try to narrow down the issue more and update if i find anything

**Comment by shreygargofficial** [comment]
Not only on keyboard or text input related things, it is occurring for
floating buttons also which are positioned absolute, If you can refer the
video attached above


*Shrey Garg8871009875*

**Comment by hirbod** [comment]
The navigation issue is caused by react native screens v4. https://github.com/software-mansion/react-native-screens/issues/2550

**Comment by shreygargofficial** [comment]
> @shreygargofficial i can repro your issue.
> 
> * it works fine on sdk 51, both new arch and old arch.
> * from sdk 52, the issue happens no matter on new arch or old arch.
> * mostly relate to keyboard. from your two videos, i cannot clearly see whether you touched at any textinput? could you help to clarify this? the jitter issue happens when i touch textinput (and keyboard appears)
> * the most trivial repro step for me is from login screen -> signup -> touch textinput -> back to login screen. as the video shown: [Screen_recording_20241205_024251.webm](https://github.com/user-attachments/assets/6cb85857-ad43-46ef-835a-027e60abd400)
> 
> will try to narrow down the issue more and update if i find anything

No, I haven't touched the keyboard in the video secondly, in the first video the layout is like there is wrapper View with flex:1 then there are 4 child in that of flex 3, 3, 7, and 3 respectively. The First and third child of flex 3 and 7 is having a scrollView in that flexed View..


In the second video there is just a floating button with position absolute that cause issue while upgrading from older version to newer

**Comment by Kudo** [comment]
> No, I haven't touched the keyboard in the video secondly, in the first video the layout is like there is wrapper View with flex:1 then there are 4 child in that of flex 3, 3, 7, and 3 respectively. The First and third child of flex 3 and 7 is having a scrollView in that flexed View..
> 
> In the second video there is just a floating button with position absolute that cause issue while upgrading from older version to newer

alright. would be good if you could record full videos from launching the app to the problems, so that we can know the all steps to repro. if the account's generated content is also relevant, would be good if you could share an account/password that could demonstrate the problems.

**Comment by shreygargofficial** [comment]
sure…Today around 1:00 Am IST i will post the video


*Shrey Garg8871009875*

**Comment by shreygargofficial** [comment]
Any update ?

**Comment by gozegotushar** [comment]
i am also facing this one, my exploration says if component content is in form of 

<ScrollView style={{flex:1}}/> or <FlatList style={{flex:1}}/>
<Bottom-Component style={{position:'absolute'}}/>

change this to : provide fix height to ScrollView and Flatlist and then align bottom-component after it, in a linear way .... dont use absolute position, else it will flicker.

**Comment by shreygargofficial** [comment]
It seems that scrollView and other component along with it can't be flexed  @Kudo @hirbod

**Comment by gozegotushar** [comment]
Yeah, with this react native update this is what I think ... absolute-bottom view is creating flicker.

I am able to resolve flickering in 7 to 8 screens using this approach.

> It seems that scrollView and other component along with it can't be flexed [@Kudo](https://github.com/Kudo) [@hirbod](https://github.com/hirbod)

**Comment by stilus-tayyipguzel** [comment]
Any update?
